packages/neuron-ui/src/components/DataSetting/hooks.ts

```diff
@@ -20,13 +20,13 @@ export const useDataPath = () => {
   const [isDialogOpen, setIsDialogOpen] = useState(false)
   const [faidMessage, setFaidMessage] = useState('')
 
-  useDidMount(() => {
+  useEffect(() => {
     getCkbNodeDataPath().then(res => {
       if (isSuccessResponse(res)) {
         setPrevPath(res.result!)
       }
     })
-  })
+  }, [network?.chain, network?.id])
```

Comment:
When the network chain switch, the ckb data directory will be changed, if we set the different IDs for different network chains, it can be removed. https://github.com/nervosnetwork/neuron/pull/2921#issuecomment-1793590460